client/mysqldump.c

```diff
@@ -3517,15 +3517,18 @@ static int dump_triggers_for_table(char *table_name, char *db_name)
 
     if (mysql_query(mysql, query_buff))
     {
-      /*
-        mysqldump is being run against old server, that does not support
-        SHOW CREATE TRIGGER statement. We should use SHOW TRIGGERS output.
-
-        NOTE: the dump may be incorrect, as old SHOW TRIGGERS does not
-        provide all the necessary information to restore trigger properly.
-      */
-
-      dump_trigger_old(sql_file, show_triggers_rs, &row, table_name);
+      if (ER_PARSE_ERROR == mysql_errno(mysql)) {
+	/*
+	  mysqldump is being run against old server, that does not support
+	  SHOW CREATE TRIGGER statement. We should use SHOW TRIGGERS output.
+
+	  NOTE: the dump may be incorrect, as old SHOW TRIGGERS does not
+	  provide all the necessary information to restore trigger properly.
+	*/
+        dump_trigger_old(sql_file, show_triggers_rs, &row, table_name);
+      } else {
```

Comment:
`else`
`{`

Reply:
Do we need any test case for this ?

Reply:
You can't have a MTR test case because that is only against the current server version where you can't generate a parse error on triggers. A buildbot test with against a pre-5.5 instance could be done, however that's outside the scope of this PR. There's no current permission denied error tests in mysql-test/main/mysqldump.test. Can you think of another way 'SHOW TRIGGERS' could generate an error?